siege_utilities/files/paths.py

```
@@ -119,19 +118,22 @@ def unzip_file_to_directory(zip_file_path: FilePath,
-                member_path = target_dir / member
-                # Resolve to absolute path and check it's within target_dir
+                # Reject absolute paths and any traversal that escapes
+                # target_dir. Path.resolve() collapses '..' segments;
+                # if the result isn't a child of target_dir, it's
+                # malicious.
                 try:
-                    member_path.resolve().relative_to(target_dir.resolve())
+                    (target_dir / member).resolve().relative_to(target_resolved)
                 except ValueError:
-                    log.error(f"Zip slip attempt detected: {member}")
+                    log.error(f"Zip slip attempt detected, skipping: {member!r}")
```

Ruff (0.15.12)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)